assets/aml-benchmark/components/src/inference_postprocessor/inference_postprocessor.py

```diff
@@ -90,129 +134,606 @@ def __post_init__(self) -> None:
+            )
+            if self.ground_truth_column_name:
+                result_df[self.ground_truth_column_name] = actual_df[
+                    self.ground_truth_column_name
+                ]
+            else:
+                result_df = actual_df
+        return result_df
+
+    def apply_find_first(self, text: str) -> str:
+        """Find and return first occurence of any candidate in the text."""
+        if self.find_first:
+            min_index = len(text)
+            first_candidate = ""
+            candidates = list(map(lambda x: x.strip(), self.find_first.split(",")))
+            for candidate in candidates:
+                index = text.find(candidate)
+                if index != -1 and index < min_index:
+                    min_index = index
+                    first_candidate = candidate
+            return first_candidate
+        return text
+
+    def apply_regex_expr(self, text: str) -> str:
+        """Apply regex on the given text."""
+        if self.regex_expr:
+            # self.regex_expr = json.loads(json.dumps(self.regex_expr))
+            self.regex_expr = self._convert_to_unicode(self.regex_expr)
+            matches = re.search(self.regex_expr, text, flags=re.DOTALL)
+            if matches is None or len(matches.groups()) == 0:
+                return text
+            return matches.group(1)
+        return text
+
+    def apply_extract_number(self, text: str, default: str = "0") -> str:
+        """Extract the first or last number from text if provided."""
+        if self.extract_number is None:
+            return text
+        number_pattern = re.compile(r"(\-?[0-9\.\,\s]+)")
+        match = number_pattern.findall(text)
+        strategy = self.extract_number
+        if match:
+            if strategy == "last":
+                match = match[::-1]
+            for m in match:
+                if not re.search(r"\d", m):
+                    # we matched with a comma or full-stop, skip this
+                    continue
+                else:
+                    m = m.strip()
+                    m = m.rstrip(".")
+                    # we only accept space and comma as separators of 3 digits in a number
+                    m = m.replace(" ", ",")
+                    m = m.strip(",")
+                    if "," in m:
+                        parts = m.split(",")
+                        if all(len(part) == 3 for part in parts[1:]):
+                            m = "".join(parts)
+                        else:
+                            m = parts[-1] if strategy == "last" else parts[0]
+                    try:
+                        # Test that the matched string is a number
+                        np.fromstring(m, sep=" ")
+                        return m
+                    except SyntaxError:
+                        # we matched with something that is not a number
+                        pass
+        return default

```

Comment:
@iamrk04 This is taken from Babel and I dare not to look into simplification right now as first, it might delay the check-ins, second, the test cases created by Babel has passed this. This is something that breaks sometimes, and they modify it. The latest change was made on Sept 25th. For now, I'm taking their experience into account kept it as it is.